robomaker-sample-app-ci/src/aws-robomaker-sample-application-ci.ts

```diff
@@ -65,127 +65,109 @@ function getExecOptions(workingDir, extraPath, listenerBuffers?): ExecOptions {
-  } catch (error) {
-    core.setFailed(error.message);
-  }
+  await exec.exec("colcon", ["build", "--build-base", "build", "--install-base", "install"], getWorkingDirExecOptions());
 }
 
 async function bundle() {
-  let delay_ms = 1000 * MINIMUM_BACKOFF_TIME_SECONDS; 
-  // indexed from 0 because COLCON_BUNDLE_RETRIES is the number of retries AFTER the initial try
-  for (let i = 0; i <= COLCON_BUNDLE_RETRIES; i++) {
-    try {
-      const bundleFilename = path.basename(WORKSPACE_DIRECTORY);
-      await exec.exec("colcon", ["bundle", "--build-base", "build", "--install-base", "install", "--bundle-base", "bundle"], getWorkingDirExecOptions());
-      await exec.exec("mv", ["bundle/output.tar", `../${bundleFilename}.tar`], getWorkingDirExecOptions());
-      await exec.exec("rm", ["-rf", "bundle"], getWorkingDirExecOptions());  // github actions have been failing with no disk space
-      break; // break if colcon bundle passes
-    } catch (error) {
-      await exec.exec("rm", ["-rf", "bundle"], getWorkingDirExecOptions()); // remove erred bundle assets
-      if (i == COLCON_BUNDLE_RETRIES){
-        core.setFailed(error.message); // set action to Failed if the colcon bundle fails even after COLCON_BUNDLE_RETRIES number of retries
-        break;
-      }
-      console.log(`Colcon bundle failed.. retrying in ${delay_ms} milliseconds`);
-      await delay(delay_ms); // wait for next retry per the current exponential backoff delay
-      delay_ms = Math.min(delay_ms * 2,  1000 * MAXIMUM_BACKOFF_TIME_SECONDS); // double the delay for the next retry, truncate if required
-    }
-  }
+  // indexed from 0 because RETRIES is the number of retries AFTER the initial try
+  const bundleFilename = path.basename(WORKSPACE_DIRECTORY);
+  await exec.exec("colcon", ["bundle", "--build-base", "build", "--install-base", "install", "--bundle-base", "bundle"], getWorkingDirExecOptions());
+  await exec.exec("mv", ["bundle/output.tar", `../${bundleFilename}.tar`], getWorkingDirExecOptions());
+  await exec.exec("rm", ["-rf", "bundle"], getWorkingDirExecOptions());  // github actions have been failing with no disk space
 }
 
 async function run() {
+  let delay_ms = 1000 * MINIMUM_BACKOFF_TIME_SECONDS;
   console.log(`ROS_DISTRO: ${ROS_DISTRO}`);
   console.log(`WORKSPACE_DIRECTORY: ${WORKSPACE_DIRECTORY}`);
   console.log(`GENERATE_SOURCES: ${GENERATE_SOURCES}`);
-  console.log(`COLCON_BUNDLE_RETRIES: ${COLCON_BUNDLE_RETRIES}`);
+  console.log(`RETRIES: ${RETRIES}`);
   
-  // check if COLCON_BUNDLE_RETRIES is valid (i.e. 0<) and not too large (i.e. <10)  
-  if (COLCON_BUNDLE_RETRIES<0 || 9<COLCON_BUNDLE_RETRIES){
-    core.setFailed(`Invalid number of colcon bundle retries. Must be between 0-9 inclusive`);
+  // check if RETRIES is valid (i.e. 0<) and not too large (i.e. <10)  
+  if (RETRIES<0 || 9<RETRIES){
+    core.setFailed(`Invalid number of retries. Must be between 0-9 inclusive`);
```

Comment:
Is setting `core.setField` enough to exit from the function?

Reply:
good point, i think it might be safer to `return` explicitly